js/enable-select2.js

```diff
@@ -9,11 +11,14 @@ CRM.$(function() {
    * we're using the "MutationObserver" to listen to DOM changes
    */
 
+  // Skips select2 for selects inside .no-select2 elements:
+  CRM.$('.no-select2 select').addClass('no-select2');
```

Comment:
Can't you somehow put this in this selector?
```js
CRM.$('select:visible:not(.no-select2):not(.crm-form-multiselect)')
```

Reply:
I tried to change the selector to this:
`*:not(.no-select2) select:visible:not(.no-select2):not(.crm-form-multiselect)`

But it didn't work. Also tried `div:not(.no-select2)` and nothing happened.

Reply:
@AkA84 I tried a few things, but even by running this single statement on the console `CRM.$('select:visible:not(.no-select2):not(.crm-form-multiselect)')` causes the selects to turn into select2 (it seems jQuery modifies the DOM when querying it???)

Another thing I noticed is that even if the jQuery DOM query returns no elements, just by appending `.select2` to the empty result causes all selects to update.

This is the best I could do, which would also add this fix to AJAX requests (unlike my first solution):

```js
var observer = new MutationObserver(debounce(function () {
    CRM.$('div:not(.no-select2) select:visible:not(.no-select2):not(.crm-form-multiselect)')
    .each(function () {
      var select = CRM.$(this);
      var noSelect2 = select.parents().filter('.no-select2').length > 0;

      if (noSelect2) {
        return;
      }

      select.select2({
        containerCss: {
          display: 'inline-block'
        }
      })
      .on('change', clearSelect2);
    });
  }, 50));
```

Should I go ahead and update the solution with this snippet?

Reply:
Looks good to me ^^

Reply:
@AkA84 this is the issue I was explaining you before:

![anim](https://user-images.githubusercontent.com/1642119/30860646-57e51f9e-a274-11e7-9a86-5b428154f04f.gif)
- the first time it breaks just by opening the work pattern tab.
- the second time it breaks when I run the jQuery DOM query. @igorpavlov could not reproduce the same issue though, we suspect it's because of OS differences.